torch_xla/csrc/view.cpp

```diff
@@ -78,7 +78,8 @@ ViewInfo::ViewInfo(xla::Shape shape, std::vector<xla::int64> sizes)
 
 ViewInfo::ViewInfo(std::vector<xla::int64> sizes,
                    std::vector<xla::int64> permutation, xla::PrimitiveType type)
-    : shape(xla::ShapeUtil::MakeShape(type, xla::Permute(permutation, sizes))),
+    : shape(xla::ShapeUtil::MakeShape(
+          type, xla::Permute(xla::InversePermutation(permutation), sizes))),
```

Comment:
@dlibenzi Are you aware of any utility function to avoid taking the inverse permutation? If not, should we add one of our own instead of doing this?

Reply:
I am not sure why we need the above.
Is the PT permutation an inverse WRT the ones defined by the XLA APIs?
The shape in ViewInfo is supposed to be the result shape (while "sizes" represents the shape of the source), so the current operation seems like the correct one.

Reply:
I've checked, `xla::Permute` is the inverse of what we need. It applies the permutation to the *output*, whereas we need to apply it to the *input*. In other words, we need to gather the sizes by `permutation` indices, not scatter them to the output at `permutation` positions.

Reply:
For example, in our permute test the input has size (2, 3, 4). If the permutation is (1, 2, 0), the correct output shape is (3, 4, 2) -- take the input sizes in the order specified by the permutation. Before this change, we returned (4, 2, 3), because `xla::Permute` uses the permutation as output indices -- 2 gets assigned to output position 1, 3 to output position 2, 4 to output position 0.

Reply:
We can have a Permute() API in XlaHelpers then.

Reply:
Done.